src/Uno.UI.Composition/Composition/Uno/PaintingSession.skia.cs

```diff
@@ -10,16 +11,19 @@ namespace Uno.UI.Composition;
 // Accessing Surface.Canvas is slow due to SkiaSharp interop.
 // Avoid using .Surface.Canvas and use .Canvas right away.
 /// <param name="RootTransform">The transform matrix to the root visual of this drawing session (which isn't necessarily the identity matrix due to scaling (DPI) and/or RenderTargetBitmap.</param>
-internal record struct DrawingSession(SKSurface Surface, SKCanvas Canvas, in DrawingFilters Filters, in Matrix4x4 RootTransform) : IDisposable
+internal readonly record struct PaintingSession(SKSurface Surface, SKCanvas Canvas, in DrawingFilters Filters, in Matrix4x4 RootTransform) : IDisposable
 {
-	public static void PushOpacity(ref DrawingSession session, float opacity)
+	[Pure]
+	public PaintingSession WithOpacity(float opacity)
 	{
 		// We try to keep the filter ref as long as possible in order to share the same filter.OpacityColorFilter
 		if (opacity is not 1.0f)
 		{
-			var filters = session.Filters;
-			session = session with { Filters = filters with { Opacity = filters.Opacity * opacity } };
+			var filters = Filters;
+			return this with { Filters = filters with { Opacity = filters.Opacity * opacity } };
 		}
+
+		return this;
```

Comment:
Cf. comment in the `Visual.skia.cs#Render` ... but I think this is dangerous: the number of `PaintingSession` (and their disposal) is linked to the number of `Save` we do on the `Canvas`. "Creating" a session should not be implciit like here. 

(Note: I used a `record` only for the light syntax, but we should consider to actually convert it to a standard `struct` to avoid the risk to implicitly create new sessions like here with the `with` keyword)